sync/fetch_queue.go

```diff
@@ -67,27 +67,36 @@ func (fq *fetchQueue) work() error {
 
 	defer fq.shutdownRecover()
 	output := fetchWithFactory(newFetchWorker(fq.workerInfra, runtime.NumCPU(), fq.batchRequestFactory, fq.queue, fq.name))
-	for out := range output {
-		fq.Debug("new batch out of queue")
-		if out == nil {
-			fq.Info("close queue")
-			return nil
-		}
+	wg := sync.WaitGroup{}
+	wg.Add(runtime.NumCPU())
+	for i := 0; i < runtime.NumCPU(); i++ {
```

Comment:
`fq.handleFetch(bjb)` is the actual work happening in these threads? I assume it's mostly CPU-bound and that's why you use `runtime.NumCPU()` as the concurrency factor?

In any case, I'd make a constant/variable for this instead of calling `runtime.NumCPU()` twice. Would make the code clearer IMO and be slightly less error prone.